packages/ra-core/src/controller/RecordContext.tsx

```diff
@@ -0,0 +1,63 @@
+import { createContext, useContext } from 'react';
+import pick from 'lodash/pick';
+import { Record } from '../types';
+
+export interface RecordContextValue {
+    record?: Partial<Record>;
+    loaded: boolean;
+    loading: boolean;
+}
+
+/**
+ * Context to store the result of the useRecord() hook.
+ *
+ * Use the useRecordContext() hook to read the context. That's what the Edit components do in react-admn.
+ *
+ * @example
+ *
+ * import { useEditController, EditContext } from 'ra-core';
+ *
+ * const Edit = props => {
+ *     const controllerProps = useEditController(props);
+ *     return (
+ *         <EditContext.Provider value={controllerProps}>
+ *             ...
+ *         </EditContext.Provider>
+ *     );
+ * };
+ */
+export const RecordContext = createContext<RecordContextValue>({
+    loaded: null,
+    loading: null,
+    record: null,
+});
+
+RecordContext.displayName = 'RecordContext';
+
+export const usePickRecordContext = <
+    ContextType extends RecordContextValue = RecordContextValue
+>(
+    context: ContextType
+) => {
+    return pick(context, ['record', 'loaded', 'loading']);
+};
+
+/**
+ * Hook to read the record from a context which provide one, such as the EditContext or ShowContext.
+ *
+ * Must be used within a <EditContextProvider> (e.g. as a descendent of <Edit>
+ * or <EditBase>) or within a <ShowContextProvider> (e.g. as a descendent of <Show>
+ * or <ShowBase>)
+ *
+ * @returns {RecordContextValue} The record context
+ */
+export const useRecordContext = () => {
+    const context = useContext<RecordContextValue>(RecordContext);
+
+    if (!context) {
```

Comment:
This will never happen. If you call `useContext` outside of the Context.Provider, you'll get the default context (in your case, the context line 29).

```jsx
import React,  { createContext, useContext } from "react";

const FooContext = createContext({ hello: 'world'})

export default function App() {
  const context = useContext(FooContext);
  return (
      <h1>Hello {context.hello}</h1>
  );
}

// always render "Hello world"
```